src/Constraint/Constraint.php

```diff
@@ -84,6 +106,42 @@ public function matches(ConstraintInterface $provider)
         return $provider->matches($this);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    public function isSubsetOf(ConstraintInterface $constraint)
+    {
+        if ($constraint instanceof EmptyConstraint) {
+            return true;
+        }
+
+        if ($constraint instanceof MultiConstraint) {
+            if ($constraint->isConjunctive()) {
+                foreach ($constraint->getConstraints() as $c) {
+                    if (!$this->isSubsetOf($c)) {
+                        return false;
+                    }
+                }
+
+                return true;
+            } else {
+                foreach ($constraint->getConstraints() as $c) {
+                    if ($this->isSubsetOf($c)) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        if ($this->operator === self::OP_NE) {
+            return (string) $constraint === (string) $this;
+        }
+
+        return $constraint->matches($this);
```

Comment:
are you sure about that ? The fact that they intersect does not make it a subset.

Reply:
See `<2.0` and `>=1.1` for an example. These constraints are intersecting, but they are not a subset of each other.

Reply:
Yeah no nothing is sure here, it was a first draft rough implementation but it is very much broken :)